src/odemis/driver/xt_client.py

```diff
@@ -2279,6 +2286,58 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
+                # not changed), resulting in misplaced tiles in the image display. Even when the stage has physically
+                # moved to the target position but took time in updating its actual position through metadata
+                # The issue is resolved by checking if only x and y stage metadata reports the target position properly
+
+                # The stage is not moving anymore, but we still want to wait until the position has been updated
+                # TODO: base the timeout on the stage movement time estimation (est. *10)
+                timeout = 10  # s
+                expected_end_time = time.time() + timeout  # s
+                check_pos = True
+                wait_duration = 20e-03  # s
+
+                # If rotation and tilt are there in target position, skip checking the metadata
+                if 'r' in pos or 't' in pos:
+                    logging.debug("Position requested to move to includes t (rx/tilt) or r (rz/rotation). "
+                                  "Reported position accuracy is not checked.")
+                    check_pos = False
+
+                # Get the target position in absolute coordinates
+                if not rel:
+                    target_pos = pos
+                else:
+                    target_pos = {ax: val + pos[ax] for ax, val in orig_pos.items()}
+
+                while check_pos:
+                    current_pos = self.parent.get_stage_position()
+                    # Every axis requested to move should have moved (compared to the position before starting)
+                    # if there is some change w.r.t starting position, proceed to check the target position
+                    if all(current_pos[a] != op for a, op in orig_pos.items()):
+                        # For x, y, and z, be extra picky and wait until they are "almost" at the target
+                        # Due to y-z linkage, there is an equivalent change in z axis when y axis is changed
+                        # Therefore, it is not important to check z
+                        # check for x and y axes as they move repeatedly in overview acquisitions
+                        axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
+                                                          axes={"x", "y"}, atol_linear=1e-6)
+
+                        if axes_updated:
+                            logging.debug("Position has updated fully: from %s -> %s", orig_pos,
+                                          current_pos)
+                            break
+
+                    if time.time() > expected_end_time:
+                        logging.warning("Stage position after move + %s s is moved to %s instead of target pos: %s. "
+                                        "Giving up waiting", current_pos, timeout, target_pos)
+                        break
+
+                    time.sleep(wait_duration)
```

Comment:
You should add a check for cancellation here, before the sleep, so that it ends early if the stage was stopped early:
```
                if future._must_stop.is_set():
                    raise CancelledError()
```